documentation/src/main/asciidoc/reference/coordination.asciidoc

```diff
@@ -0,0 +1,508 @@
+have a look at the <<architecture-examples>>.
+====
+
+An application using Hibernate Search usually relies on multiple threads,
+or even multiple application instances,
+which will update the database concurrently.
+
+The coordination strategy defines how these threads/nodes will coordinate with each other
+in order to update indexes according to these database updates,
+in a way that ensures consistency,
+prevents data loss,
+and optimizes performance.
+
+This strategy is set through configuration properties:
+
+[source]
+----
+hibernate.search.coordination.strategy = database-polling
+----
+
+The default for this property is `none`.
+
+See the following subsections for details about available strategies.
+
+[[coordination-none]]
+== No coordination
+
+[[coordination-none-basics]]
+=== Basics
+
+The default strategy is the simplest and does not involve any additional infrastructure
+for communication between application nodes.
+
+All <<mapper-orm-indexing-automatic,automatic indexing>> operations
+are executed directly in application threads,
+which gives this strategy the unique ability to provide <<mapper-orm-indexing-automatic-synchronization,synchronous indexing>>,
+at the cost of a few limitations:
+
+* <<limitations-parallel-embedded-update>>
+* <<limitations-backend-indexing-error>>
+
+include::components/coordination-comparison-note.asciidoc[]
+
+[[coordination-none-indexing]]
+=== How indexing works without coordination
+
+[[coordination-none-indexing-detected-changes]]
+Changes have to occur in the ORM session in order to be detected::
+[[mapper-orm-indexing-automatic-concepts-changes-in-session]] Hibernate Search uses internal events of Hibernate ORM in order to detect changes:
+these events will only be triggered if you actually manipulate managed entity objects in your code,
+updating them by setting their properties or deleting them by calling the appropriate method
+on the Hibernate ORM session.
++
+Conversely, changes resulting from `insert`/`delete`/`update` queries, be it SQL or JPQL/HQL queries,
+are not detected by Hibernate Search.
+This is because queries are executed on the database side,
+without Hibernate having any knowledge of which entities are actually created, deleted or updated.
+One workaround is to explicitly reindex after you run such queries,
+either using the <<mapper-orm-indexing-massindexer,`MassIndexer`>>
+or <<mapper-orm-indexing-manual,manually>>.
+[[coordination-none-indexing-change-filter]]
+Only relevant changes trigger indexing::
+[[mapper-orm-indexing-automatic-concepts-changes-filter]] Hibernate Search is aware of the properties that are accessed when building indexed documents.
+Thanks to that knowledge, it is able to skip reindexing
+when a property is modified, but does not affect the indexed document.
++
+You can control this "dirty checking" by setting
+the <<configuration-property-types,boolean property>> `hibernate.search.automatic_indexing.enable_dirty_check`:
```

Comment:
is this property supported?

Reply:
Yes? See `org.hibernate.search.mapper.orm.session.impl.ConfiguredAutomaticIndexingStrategy#DIRTY_CHECK_ENABLED`.

It's been here for a long time, since Search 5 and probably before. Personally I don't see why someone would use it, but I kept it in 6.0 to ease migration.